test/clj/game/cards/upgrades_test.clj

```diff
@@ -3336,3 +3336,243 @@
+    (play-from-hand state :corp "K. P. Lynn" "New remote")
+    ))
+
+(deftest fractal-threat-matrix
+  (do-game
+    (new-game {:corp {:hand ["Fractal Threat Matrix" "Najja 1.0"]}
+               :runner {:deck [(qty "Acacia" 7)]}})
+    (play-from-hand state :corp "Fractal Threat Matrix" "New remote")
+    (play-from-hand state :corp "Najja 1.0" "Remote 1")
+    (take-credits state :corp)
+    (is (= 2 (count (:deck (get-runner)))))
+    (run-on state :remote1)
+    (let [najja (get-ice state :remote1 0)
+          ftm (get-content state :remote1 0)]
+      (rez state :corp najja)
+      (rez state :corp ftm)
+      (run-continue state)
+      (card-side-ability state :runner najja 0)
+      (click-prompt state :runner "End the run")
+      (card-side-ability state :runner najja 0)
+      (click-prompt state :runner "End the run")
+      ;; All subroutines broken. Manually trigger Fractal Thread Matrix
+      (card-ability state :corp ftm 0)
+      (is (zero? (count (:deck (get-runner)))))
+      (is (= 2 (count (:discard (get-runner)))))
+      )))
+
+(deftest shell-corporation
+  (do-game
+    (new-game {:corp {:hand ["Shell Corporation"]}})
+    (play-from-hand state :corp "Shell Corporation" "New remote")
+    (let [sc (get-content state :remote1 0)]
+      (rez state :corp sc)
+      (card-ability state :corp (refresh sc) 0)
+      (card-ability state :corp (refresh sc) 0)             ;;Retrigger ability, no extra credits are given
+      (take-credits state :corp)
+      (take-credits state :runner)
+      (card-ability state :corp (refresh sc) 0)             ;;Store another 3c. Total 6c
+      (take-credits state :corp)
+      (take-credits state :runner)
+      (card-ability state :corp (refresh sc) 1)             ;;Now take those 6c
+      (is (= 12 (:credit (get-corp)))))))
+
+(deftest panic-button
+  (do-game
+    (new-game {:corp {:hand ["Panic Button"] :deck ["Enigma"]}})
+    (play-from-hand state :corp "Panic Button" "HQ")
+    (let [panic-btn (get-content state :hq 0)]
+      (rez state :corp panic-btn)
+      (take-credits state :corp)
+      (run-on state :hq)
+      (card-ability state :corp (refresh panic-btn) 0)
+      (is (= 1 (count (:hand (get-corp)))))
+      (is (zero? (count(:deck (get-corp))))))))
+
+(deftest expo-grid
+  (do-game
+    (new-game {:corp {:hand ["Expo Grid" "Dedicated Response Team" "Breaking News"]}})
+    (play-from-hand state :corp "Dedicated Response Team" "New remote") ;;-2c
+    (play-from-hand state :corp "Expo Grid" "Remote 1")
+    (let [drt (get-content state :remote1 0)
+          expo (get-content state :remote1 1)]
+      (rez state :corp drt)
+      (rez state :corp expo)
+      (take-credits state :corp)                            ;;+1c
+      (let [total-corp-credits (:credit (get-corp))]
+        (take-credits state :runner)
+        (is (= (+ 1 total-corp-credits) (:credit (get-corp))) "Corp gains 1c") ;;+1c
+        ;;Replace asset with agenda
```

Comment:
Instead of doing both checkss in a single test, create a second test and try it there. Reduces cognitive overhad.